src/Offer/CommandHandlers/DeleteOrganizerHandler.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CultuurNet\UDB3\Offer\CommandHandlers;
+
+use Broadway\CommandHandling\CommandHandler;
+use CultuurNet\UDB3\Offer\Commands\DeleteOrganizer;
+use CultuurNet\UDB3\Offer\OfferRepository;
+use CultuurNet\UDB3\UiTPAS\Validation\EventHasTicketSalesGuard;
+
+final class DeleteOrganizerHandler implements CommandHandler
+{
+    private OfferRepository $offerRepository;
+
+    private EventHasTicketSalesGuard $organizerChangeAllowedBasedOnTicketSales;
+
+    public function __construct(
+        OfferRepository $offerRepository,
+        EventHasTicketSalesGuard $organizerChangeAllowedBasedOnTicketSales
+    ) {
+        $this->offerRepository = $offerRepository;
+        $this->organizerChangeAllowedBasedOnTicketSales = $organizerChangeAllowedBasedOnTicketSales;
+    }
+
+    public function handle($command): void
+    {
+        if (!($command instanceof DeleteOrganizer)) {
+            return;
+        }
+
+        $this->organizerChangeAllowedBasedOnTicketSales->guard($command);
```

Comment:
Maybe we should do the checks if there is a change in these new command handlers and not in `EventHasTicketSalesGuard`, since it is dependent on the command? That way `EventHasTicketSalesGuard` does not need to know about every command that needs to do this check, it could just check if there are ticket sales or not.

Reply:
Would it be safe to always call `eventHasTicketSales` even for places? Then the amount of code from `EventHasTicketSalesGuard` can be reduced and there won't be much duplication in the command handlers.

Reply:
I meant to move [this check](https://github.com/cultuurnet/udb3-silex/pull/1055/files#diff-289ee1aff94dbd0d438f353e6b00b25bc0d75f221b309ddd0217f66112bc1752R51-R55) to UpdateOrganizerHandler, and [this check](https://github.com/cultuurnet/udb3-silex/pull/1055/files#diff-289ee1aff94dbd0d438f353e6b00b25bc0d75f221b309ddd0217f66112bc1752R56-R59) to DeleteOrganizerHandler.
That way `EventHasTicketSalesGuard` doesn't kneed to know in which contexts it will be called, and we can re-use it in other command handlers later if ever needed.

Reply:
> Would it be safe to always call eventHasTicketSales even for places?

I think if you'd call it for a place id it will result in an exception, but those are caught and handled as if there are no ticket sales so it could work. We should test it (manually) to be sure though.

Reply:
Good proposal, this should be fixed now.